paddle/fluid/framework/details/all_reduce_op_handle.h

```diff
@@ -46,13 +49,17 @@ class AllReduceOpHandle : public OpHandleBase {
 
  protected:
   void RunImpl() override;
-
   std::vector<Scope *> local_scopes_;
+
+#if !(defined(PADDLE_WITH_CUDA) && !defined(_WIN32))
+  // NCCLOpHandleBase already have these attributes.
+  // Will polish it by class inheritance framework.
   std::vector<platform::Place> places_;
+#endif
```

Comment:
If we remove `NCCLOpHandleBase `, the above fix is unnecessary.

Reply:
TODO: https://github.com/PaddlePaddle/Paddle/issues/17513